packages/xml-views-validation/test/validators/element/non-stable-id-spec.ts

```diff
@@ -0,0 +1,172 @@
+    });
+
+    it("will detect missing stable in sub element when the element has attribute sap.ui.dt:designtime='not-adaptable'", () => {
+      assertSingleIssue(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons"
+          xmlns:sap.ui.dt="sap.ui.dt">
+            <m:Panel sap.ui.dt:designtime="not-adaptable">
+              <🢂m:Button🢀></m:Button>
+            </m:Panel>
+        </mvc:View>`,
+        getMessage(NON_STABLE_ID, "Button")
+      );
+    });
+  });
+
+  context("negative edge cases", () => {
+    let assertNoIssues: (xmlSnippet: string) => void;
+    before(() => {
+      assertNoIssues = partial(assertNoIssuesBase, ui5SemanticModel, {
+        element: [validateNonStableId],
+      });
+    });
+
+    it("will not detect an issue when there is a root-whitelisted UI5 class", () => {
+      assertNoIssues(
+        `<!-- sap.ui.core.mvc.View is whitlisted root class -->
+        <mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons">
+        </mvc:View>`
+      );
+    });
+
+    it("will not detect an issue when there is a root-whitelisted UI5 class - core ns", () => {
+      assertNoIssues(
+        `<!-- sap.ui.core.FragmentDefinition is whitlisted root class -->
+        <core:FragmentDefinition xmlns="sap.m" xmlns:core="sap.ui.core">
+        </core:FragmentDefinition>`
+      );
+    });
+
+    it("will not detect an issue when there is a whitelisted UI5 class", () => {
+      assertNoIssues(
+        `<mvc:View xmlns:core="sap.ui.core" xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons">
+            <m:Button id="yyy">
+                <m:customData>
+                    <core:CustomData></core:CustomData>
+                </m:customData>
+            </m:Button>
+        </mvc:View>`
+      );
+    });
+
+    it("will not detect an issue when the control has attribute attribute sap.ui.dt:designtime='not-adaptable'", () => {
+      assertNoIssues(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+            xmlns:mvc="sap.ui.core.mvc"
+            xmlns="sap.ui.commons"
+            xmlns:sap.ui.dt="sap.ui.dt">
+              <m:Panel sap.ui.dt:designtime="not-adaptable">
+              </m:Panel>
+        </mvc:View>`
+      );
+    });
+
+    it("will not detect an issue when the control has attribute attribute sap.ui.dt:designtime='not-adaptable'", () => {
+      assertNoIssues(
+        `<mvc:View xmlns:uxap="sap.uxap" xmlns:m="sap.m"
+          xmlns:mvc="sap.ui.core.mvc"
+          xmlns="sap.ui.commons"
+          xmlns:sap.ui.dt="sap.ui.dt">
+            <m:Panel sap.ui.dt:designtime="not-adaptable">
+            </m:Panel>
+        </mvc:View>`
+      );
+    });
+
+    it("will not detect an issue when the control has attribute attribute sap.ui.dt:designtime='not-adaptable-tree'", () => {
```

Comment:
- attribute word * 2
- emphasize it would not detect issues for **sub-elements**